diff --git a/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts b/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts
index c186961aa..9edc7c642 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts
@@ -315,6 +315,7 @@ &eMM {
 
 &avb {
     status = "okay";
+    phy-mode = "mii";
 };
 
 &pwm0 {
