diff --git a/source/1st_loader/src/evk_HWSetup.c b/source/1st_loader/src/evk_HWSetup.c
index 8ab0baa..c1a2d28 100755
--- a/source/1st_loader/src/evk_HWSetup.c
+++ b/source/1st_loader/src/evk_HWSetup.c
@@ -40,6 +40,11 @@ void InitPFC(void)
 {
     /* Set Pin function to PFC unit */
     Set_Eachpin();
+    
+    /* set control GBE RST and RST OUT to LOW */
+    PFC_WriteReg(PFC_P11_EN_MSK,  0x00FF0000);
+    PFC_WriteReg(PFC_P11_GPIO_OE, 0x00030003);
+    PFC_WriteReg(PFC_P11_GPIO_DO, 0x00030000);
 }
 
 
