diff --git a/drivers/net/phy/mscc.c b/drivers/net/phy/mscc.c
index d1a643cf5a..ffc566e0fa 100644
--- a/drivers/net/phy/mscc.c
+++ b/drivers/net/phy/mscc.c
@@ -1575,6 +1593,16 @@ static struct phy_driver VSC8584_driver = {
 
 int phy_mscc_init(void)
 {
+	/* enable PHY */
+	printf("Enable PHY 8541\n");
+
+	/* Enable GBE PHY, Deassert RST OUT# */
+	*(volatile unsigned int*)(0xB6250000 + 0x2F4) = 0x00FF0000;
+	*(volatile unsigned int*)(0xB6250000 + 0x2C4) = 0x00030003;
+	*(volatile unsigned int*)(0xB6250000 + 0x2C0) = 0x00030003;
+	/* wait 100ms */
+	CMN_DelayInUSec(100*1000);
+
 	phy_register(&VSC8530_driver);
 	phy_register(&VSC8531_driver);
 	phy_register(&VSC8540_driver);
